Fnal develop merge

gartung opened the pull request.

zackgalbreath committed:
Avoid UnboundLocalError

Make sure cdash_phase is defined before referring to it.

zackgalbreath committed:
Allow more customization for CDash reporter

Add new command line arguments to `spack install` that allow users
to set the build name, site name, and track in their CDash report.

zackgalbreath committed:
Get buildId from CDash at submit time

Pass extra data when submitting to CDash.  If CDash responds with a buildId,
construct and display helpful URL that links to the uploaded report.

scottwittenburg committed:
Make sure to urlencode cdash submit parameters

zackgalbreath committed:
Report current git commit of Spack to CDash

When using the CDash reporter, upload a Update.xml file that
indicates the hash of Spack's current git commit.

zackgalbreath committed:
Include accurate build time in CDash report

becker33 committed:
Added test for multimethod inheritance

tgamblin committed:
remove spurious print in zlib patch method

junghans committed:
pfunit: add mpi-test patch (#9387)

matthiasdiener committed:
openblas: add version 0.3.4 (#10016)

tgamblin committed:
spec: refactor and clean up Spec initialization

- Since early Spack versions, the SpecParser has (weirdly) been
  responsible for initializing Spec fields.

- This refactors initialization to take place in Spec.__init__, as it
  probably should have originally.

- This makes the code easier to read, the parser easier to understand,
  and removes the use of __new__ in the parser to initialize the Spec.

- This also makes it possible to make a completely empty Spec with
  `Spec()` -- this is an abstract Spec that will match anything.

tgamblin committed:
multimethod uses Spec() instead of parse_anonymous_spec()

- simplify logic in multimethod
- remove the requirement of multimethod invocations to walk up the stack.

tgamblin committed:
tests: fully parametrize multimethod test

- all multimethod tests are now run for both `multimethod` and
  `multimethod-inheritor`

- do this with a parameterized fixture (pkg_name) that runs the same
  tests on both

Commit:
Rewrite Rust package (#9998)

becker33 committed:
fix MRO for multimethod.__call__ using iterative algorithm.

Add tests MRO for inherited multimethods with multiple inheritance
Add tests for inherited and overridden multimethods

matthiasdiener committed:
cloc: add version 1.80 (#10029)

Commit:
Fix doxygen package (#10021)

erimar77 committed:
DIAMOND: Add version 0.9.23 (#10030)

becker33 committed:
multimethod: slight refactoring, documentation for code review

naromero77 committed:
Quantum ESPRESSO update - Nov 2018 (#9853)

* Add mising QE releases.

* Update QE dependencies and conflicts.

* QE 6.3 install method requires a patch to work properly.

* include QE git develop branch

* QE HDF5 needs Fortran support. HDF5 versions prior to 1.8.16 leads to QE runtime errors.

* Number of MKL related conflicts for QE.

* Flake8 fixes for QE for hdf5 lines.

* Pass in all compilers properly to QE configure

* Fixed external BLAS/LAPACK linkage in QE.

* Library search was problematic in QE.

* Add section headings to QE package.

* External FFTW compatible libraries properly detected in QE.

* Flake8 fixes for QE

* QE configure defaults to False for ELPA, thus Spack package should as well.

* QE depends on FFTW3 provider instead of FFTW explictly.

* Primary QE repo is gitlab.com, not github.com. Github mirror lags behind gitlab site.

* Support both 6.3-backports branch and develop branch in QE.

* QE patch needed for proper MKL detection. MKL_ROOT is detected and preferred, still falls back to finding first MKL installation in /opt/intel.

* Use FFTW-API provider versioned interface for specify QE FFT dependency.

* QE FFTW conflicts no longer need thanks to FFTW-API provider versioned interface.

* Flake8 3.6.0 fix in support of https://github.com/spack/spack/pull/9853

* Fix QE conflict with ilp64 variant in support of https://github.com/spack/spack/pull/9853

* Remove custom url_for_version for QE package and use builtin url instead. In support of https://github.com/spack/spack/pull/9853

* Use FFTW builtin flags for QE instead of custom Python code. In support of https://github.com/spack/spack/pull/9853

* Fix typo in QE package comments. In support of https://github.com/spack/spack/pull/9853

* Future proof QE package patches in anticipation of QE >= 6.3.1. In support of https://github.com/spack/spack/pull/9853

jrood-nrel committed:
Adding patch to remove unicode character from pyrope readme file (#9988)

* Adding new versions of py-rope.

balay committed:
superlu-dist: update to version 6.1.0 (#10037)

michaelkuhn committed:
cdo: fix URLs (#10025)

Commit:
clp: new package at 1.16.11 (#10040)

mamelara committed:
Spades: Add version 3.13.0 (#10033)

* Spades: Add version 3.13.0

In version 3.13.0 the CmakeLists.txt dir is at a different location than
in other versions. Versions up to 3.12.0 are located in "src" whereas
3.13.0 is a level deeper at "assemblers/src".

* Remove git and add new version hash

Commit:
r-blockmodeling: new package at 0.3.1 (#10044)

Commit:
r-sseq: new package at 1.20.0 (#10042)

Commit:
scallop: new package at 0.10.3 (#10041)

Commit:
pbsuite: new package at 15.8.24 (#10059)

jameshclrk committed:
gcc: Add patch to build GCC with glibc 2.28 and later (#10046)

ax3l committed:
Fix Catch2 Single Header (#10022)

Fix the install of the single header amalgate in catch2.

The public API for catch < 2.3.0 was "#include <catch.hpp>".
For 2.3.0+ it is "#include <catch2/catch.hpp>"

Commit:
r-expint: new package at 0.1-5 (#10043)

matthiasdiener committed:
eztrace: add new package (#10060)

cwsmith committed:
pumi: shared variant (#10057)

bhatele committed:
TraceR: add release tag/version (#10054)

jrood-nrel committed:
OpenMPI: adding static libraries with slurm support for latest versions. (#10055)

matthiasdiener committed:
gcc: add versions 7.4, 6.5 (#10061)

payerle committed:
metis: Fix issues with gcc6 and misleading indentation (#10058)

Added patch to GKlibSystem.cmake to disable warnings (which get
promoted to errors) due to -Wmisleading-indentation being added
to -Wall in gcc 6

matthiasdiener committed:
cloc: URLs are now consistent (#10068)

Fixed in upstream:
https://github.com/AlDanial/cloc/issues/352

likask committed:
mofem: add new releases and minor changes (#9138)

adamjstewart committed:
Make downloads from list_url deterministic (#10047)

bvanessen committed:
Update LBANN generator and bug fix (#9906)

* Added a package for the MDAnalysis toolkit.

* Added Ninja as the generator for LBANN, Hydrogen, and Aluminum.  Also
fixed a bug in how the Conduit variant is included in LBANN.  Added
new versions for Hydrogen and LBANN.

* Updated lbann package so that the CMAKE parameters are set when the
variant is on and off.

zackgalbreath committed:
Performance improvements for CDash reporter

* Record build output as an array of lines rather than concatenating to a
  single large string.
* Use string.find to avoid running re.search on every line of output.

SteVwonder committed:
flux-core & flux-sched: fix compilation and dependencies (#10049)

* flux-sched: fix compilation errors

configure was being run twice becuase the staging symlinks were
confusing autotools

mirror flux-core in building in a subdirectory to avoid other common
compilation problems, like the one in PR #3543

* flux-core: update dependencies

munge is no longer a dependency

provide hints for lua version so that `spack install flux-core` works
"out-of-the-box". Currently, when no explicity version is provided, the
concretizer attempts to use lua@5.3, which is invalid. Closes #10000

mwkrentel committed:
gcc: fix sys_ustat.h patch for gcc 4.8 and 4.9 (#10081)

The sys_ustat.h.patch to file sanitizer_platform_limits_posix.cc from
PR #10046 does not apply cleanly to gcc 4.8 or 4.9 (or earlier).

GCC up to 4.8.x either don't have libsanitizer or else don't include
ustat.h in sanitizer_platform_limits_posix.cc.

GCC 4.9.x includes ustat.h, but needs a slightly different patch.

The patch applies to GCC 5.x up to 6.4, and 7.x up to 7.3 and also
8.1.0.

The patch is already included in the tar files for gcc 6.5.0, 7.4.0
and 8.2.0.

gonsie committed:
added sha256 for scr v1.2.0-2 (#10082)

erimar77 committed:
LEMON: Create new package (#10077)

davydden committed:
boost: add 1.69.0 (#10083)

signals (v1) was removed in 1.69.0

junghans committed:
lammps: add new stable version (#10086)

jrood-nrel committed:
Add CUDA_HOME variable to cuda module file. (#10088)

balay committed:
superlu-dist: update v6.1.0 checksum (#10078)

erimar77 committed:
trf: add version 4.07b and install fixes (#10093)

* trf: add version 4.07 and install fixes

* suggested fixes

Miles-Perry committed:
Feature/spm (#10053)

* extract spm12.ctf to get binaries and libraries

* flake8 edits

* flake8 edits

* spm: updates

* spm: previous version

matthiasdiener committed:
cmake: add version 3.13.2 (#10092)

adamjstewart committed:
Add new packages for ELSD and ELSDc (#10072)

gonsie committed:
make dtcmp an autotools spackage (#10095)

* make dtcmp an autotools spackage

* better code

adamjstewart committed:
Add latest version of meson (#10104)

becker33 committed:
Fix spack package inheritance for module variables (#10097)

* we weren't properly setting module variables for the root package in a DAG -- just for transitive dependencies.

alalazo committed:
Turn on verbose output for MPICH builds in Travis (#10105)

Lately many CI runs for PRs are failing due to the `mpich` build that
times out on Travis (10 mins. without output). As the timeout seems to
happen consistently during the build phase, increasing the verbosity of
that test can help working around the issue.

alalazo committed:
Updated the Travis environment to xenial (#10090)

* Updated the Travis environment to xenial.
* Python 2.6 needs to remain on Trusty (can't be installed from Xenial)

adamjstewart committed:
Add latest version of cairo (#10102)

adamjstewart committed:
Add latest version of poppler (#10101)

adamjstewart committed:
Add latest version of libxslt (#10098)

michaelkuhn committed:
qt: fix openssl dependency for older versions (#10106)

tgamblin committed:
env: all commands that disambiguate specs should be env-aware

- some commands were missed in the rollout of spack environments

- this makes all commands that need to disambiguate specs restrict the
  disambiguation to installed packages in the active environment, as
  users would expect

AndrewGaspar committed:
Updates to Google Test: (#10079)

- Add version 1.8.1
- Change to sha256 checksums

Commit:
clp: add explicit datadir (#10109)

baberlevi committed:
pbsuite: pythonpath should be prepended not set (#10113)

adamjstewart committed:
Add latest version of libatomic-ops (#10099)

adamjstewart committed:
Add latest version of bdw-gc (#10100)

adamjstewart committed:
Add latest version of libsigcpp (#10103)

junghans committed:
.travis.yml: enable ccache (#6370)

adamjstewart committed:
spack pkg add: add help message (#10116)

Commit:
r-xnomial: new package at 1.0.4 (#10111)

* r-xnomial: new package at 1.0.4

* r-xnomial: add list_url

Commit:
r-popgenome: new package at 2.6.1 (#10108)

alalazo committed:
Store ccache directory explicitly in Travis. (#10119)

* Store ccache directory explicitly in Travis.

Despite we started using ccache on `develop`, it seems the cache itself
is not stored from one CI build to the next. This might be du to the
fact that our language on Travis is Python and not C nor C++.

Hence here we store the ccache directory explicitly.

Commit:
r-multitaper: new package at 1.0-14 (#10110)

matthiasdiener committed:
env setup: support --help for cd/env commands (#10069)

scheibelp committed:
mirrors: patches are now properly added to mirrors (#8993)

* This fixes a number of bugs:

  * Patches were not properly downloaded and added to mirrors.

  * Mirror create didn't respect `list_url` in packages

  * Update the `spack mirror` command to add all packages in the
    concretized DAG (where originally it only added the package specified
    by the user). This is required in order to collect patches that are specified
    by dependents. Example:
      * if X->Y and X requires a patch on Y called Pxy, then Pxy will only
        be discovered if you create a mirror with X.

  * replace confusing --one-version-per-spec option for `spack mirror create`
    with --versions-per-spec; support retrieving multiple versions for 
    concrete specs

* Implementation details:

  * `spack mirror create` now uses regular staging logic to download files 
    into a mirror, instead of reimplementing it in `add_single_spec`.

  * use a separate resource caching object to keep track of new
    resources and already-existing resources; also accepts storing
    resources retrieved from a cache (unlike the local cache)

  * mirror cache object now stores resources that are considered
    non-cachable, like (e.g. the tip of a branch);

  * the 'create' function of the mirror module no longer traverses
    dependencies since this was already handled by the 'mirror' command; 

  * Change handling of `--no-checksum`:

    * now that 'mirror create' uses stages, the mirror tests disable
      checksums when creating the mirror

    * remove `no_checksum` argument from library functions - this is now
      handled at the Spack-command-level (like for 'spack install')

Commit:
packages/phist: new version 1.7.5 (#10120)

tgamblin committed:
patches: clean up patch.py, directives, and package class properties

- cleanup patch.py:
  - make patch.py constructors more understandable
  - loosen coupling of patch.py with package

- in Package: make package_dir, module, and namespace class properties

  - These were previously instance properties and couldn't be called from
    directives, e.g. in patch.create()

  - make them class properties so that they can be used in class definition

  - also add some instance properties to delegate to class properties so
    that prior usage on Package objects still works

adamjstewart committed:
Add latest version of atk (#10125)

lingnanyuan committed:
mark: Create new package. (#7069)

* mark: Create new package.

* mark: change description.

* mark: change description.

* mark: Delete set_up environment.

* mark: replace join_path(prefix.bin, mark) with prefix.bin.mark

tz-rrze committed:
zoltan: rely on MPI wrappers instead of guessing MPI libraries (#8986)

* zoltan: only add gussed MPI libs if no MPI wrappers are used

- if MPI-wrappers are used for compilation, we can assume that
  linking works without manually specifying MPI libs
  (guessing may result in wrong libs, cf. #8979)
- thus, only guess the NPI libs and add them explicitly if no
  MPI-wrappers are used
- use llnl.util.filesystem.find_libraries instead of a locally
  defined routine to guess the MPI libs if needed
  (cf. #8979)

* zoltan: rely on MPI-wrappers to know the required MPI libs

rmsds committed:
GDL: python integration fixes (#9936)

* GDL: python integration fixes

* renamed python-related variants to follow the convention
* building the Python module requires patches currently targetting 0.9.8 othwerwise asking for the Python module *only* builds the Pyhton module
* building the python module also requires patching the vendored (with the GDL) antlr to be built as a shared library

* Typo

Co-Authored-By: rmsds <ricardo.d.silva@gmail.com>

* Rename embed-python variant to embed_python

gsjaardema committed:
SEACAS: Add preliminary support for seacas package (#9972)

* SEACAS: Add preliminary support for seacas package

* Eliminate leftover from package I copied from.

* And remove another stray line

Sorry, thought I had checked this better...

* SEACAS: Shorten long lines; fixup indentation

* SEACAS: Remove unused os import

* Add PLASMA version 18.11.0 and Lua variant (#9689)

* Add PLASMA 18.11.1 with optional Lua and test for headers

* New versions for a few python modules (#9969)

* py-jupyter-console: Add version 5.2.0

* py-jupyter-core: add version 4.4.0

* py-lxml: add version 4.2.5

* py-zmq: add version 17.1.2

* py-terminado: add version 0.8.1

* py-pexpect: add version 4.6.0

* py-pytables: add version 3.4.4

* SEACAS: Default to mpi on

* SEACAS: Modify to reduce variants

Instead of using an `allpkgs` or explicitly specifying each individual subset,
changed to minimize variants:
 * By default, the exodus and IOSS libraries are built and a few
   executables directly related to the IOSS library.
 * variant `common` defaults to True and builds the "common" seacas
   applications.  This is based on my definition, so there may be
   could be some discussion on what belongs here.
 * variant `legacy` defaults to True and builds the remainder of the
   seacas applications.  These are still used, but mainly in more
   esoteric areas.  Might be better to default to False, but I wanted
   default to just build everything which is what other SEACAS builds
   always do...

Removed the `kokkos` variants for now as I need to do some more
thinking on how best to support this.  Not often used currently,
so shouldn't be an issue.

Cleaned up other issues raised by Adam and found by me.

* SEACAS: Address flake8 issues

* SEACAS: Clean up variant package logic

Fixed package variant logic.
Consolidated some other variant handling logic
Other minor refactors for hopefully better readability

* SEACAS: Fix flake8 error

Missing whitespace around operator fixed.

olesenm committed:
ENH: fixes and updates for openfoam-com build (#10065)

- previous removal of FOAM_EXT_LIBBIN (ThirdParty) from the environment
  during the build could lead to the system scotch library being found
  instead of the spack installation.

  Avoid this largely cosmetic change to the setting in favour of
  improved build robustness. These modifications will be largely
  superfluous in the 1812 release anyhow.

- Split the patching algorithm into 'regular' and '@:1806' since the
  next versions of openfoam will again need fewer patches.

- Add hook into foamCreateManpage to be used where available

adamjstewart committed:
Add patch to fix windows.h include in freetype (#10075)

* Add patch to fix windows.h include in freetype

* Fix patch

adamjstewart committed:
Update Spack's Bash completion script (#10114)

* Update Spack's Bash completion script

* spack mirror create options changed

adamjstewart committed:
spack buildcache: --allow_root -> --allow-root (#10115)

adamjstewart committed:
Add help messages to spack gpg subcommands (#10117)

adamjstewart committed:
Add latest version of libbsd (#10121)

adamjstewart committed:
Add missing libbsd dependency to libxdmcp (#10122)

adamjstewart committed:
Add latest version of ghostscript (#10123)

adamjstewart committed:
Add latest version of gdk-pixbuf (#10126)

tgamblin committed:
Revert "mark: Create new package. (#7069)" (#10131)

This reverts commit 90b34c96d6a255489ccc03c62f64c191130a32ac.

adamjstewart committed:
Add missing gettext dependency to at-spi2-core (#10124)

ohm314 committed:
Added RocksDB package (#10084)

* Added rocksdb package

* Fixes based on @tristan0x and @pramodk comments

* Cleanup

* A couple minor fixes based on @adamjstewart 's advice

* Fixed flake8 errors

matthiasdiener committed:
git: add version 2.20.1 (#10130)

Commit:
Add spack package for saga-gis (#10018)

* Add spack package for saga-gis

* Adjusted license header

* Adjust saga-gis package, added fixme comments

* Removed version constraint

* Fixed fixme comment

daveh86 committed:
Add thread count variant to atlas (#9999)

* Add thread count variant

Atlas automatically configures itself and optimizes to run with a fixed number of threads. This can be overridden during compile by setting a flag and atlas will tune itself to use the specified number, rather than all system CPUs. 

Default value is the existing logic, autoconfigure.

* Fix Linting

* Another lint for longline

bebosudo committed:
Add variant to ecflow to choose whether to use static boost libraries. (#9933)

When this variant is set, it requires boost to be built with the +pic variant.

adamjstewart committed:
Add latest version of libxkbcommon (#10134)

Commit:
clp: add build_directory, remove old hack (#10112)

* clp: add build_directory, remove old hack

* clp: use spack-build for building

adamjstewart committed:
Add pdf variant to cairo (#10135)

matz-e committed:
flatbuffers: add 1.10.0 (#10137)

Also unpin the version from the arrow package.

gartung committed:
Revert "Flake8 fixes."

This reverts commit f744cd75143801785afd43fda7b3b7b92d96ca6f.

gartung committed:
Revert "Environment export improvements."

This reverts commit ba1e97ab7e0dd14b195ce4e656a92b5e28e612bd.

Conflicts:
	lib/spack/spack/util/environment.py

gartung committed:
Revert "--dump option to spack env to write the environment as a source-able script."

This reverts commit bb9359c5e3e1670c09ef24fc965b463b19aed5e9.

gartung committed:
Revert "Provide facility for deprioritizing system paths in a PATH-like environment variable, and for pruning duplicates."

This reverts commit e242530d9113763c37bc73ad00631b4dd0fb938c.

Conflicts:
	lib/spack/spack/util/environment.py

gartung committed:
Merge remote-tracking branch 'upstream/develop' into fnal-develop

Conflicts:
	lib/spack/spack/schema/config.py
	lib/spack/spack/test/conftest.py
	lib/spack/spack/test/environment_modifications.py
	lib/spack/spack/util/environment.py
	var/spack/repos/builtin/packages/sqlite/package.py

gartung committed:
recover changes wiped out by merge and reverts

gartung committed:
add back extensions.py

gartung committed:
Merge remote-tracking branch 'llnl/develop' into fnal-develop-m2

Conflicts:
	lib/spack/spack/cmd/env.py
	lib/spack/spack/environment.py
	lib/spack/spack/schema/config.py
	lib/spack/spack/test/conftest.py
	lib/spack/spack/test/environment_modifications.py
	lib/spack/spack/util/environment.py
	var/spack/repos/builtin/packages/sqlite/package.py

gartung committed:
fixups from merge

gartung committed:
Merge branch 'fnal-develop-m2' into fnal-develop-merge

Conflicts:
	lib/spack/spack/test/conftest.py

greenc-FNAL commented:
You can always close, force-push to your branch and then reopen. You don't need multiple PRs.

gartung committed:
all virtual packages must have a provider defined

gartung committed:
make environment dump test match the changed output of build_env dump

dpgrote committed:
WarpX: Added dependency on fftw when +psatd (#10144)

gonsie committed:
fix for mpifileutils package (#10142)

* proper include & linking of libarchive

* due to linux headers, mfu can't build on mac

* hard code deps in configure line

rather than relying on pkg-config

* +xattr during config

* must also link the lib

* find mfu.h

* long lines

adamjstewart committed:
Add additional info to MesonPackage docs (#10133)

* Add additional info to MesonPackage docs

* No Pygments lexer for Meson code, default to none

alalazo committed:
Added labels in github issue templates (#10128)

neilflood committed:
mapserver: new package (#10067)

michaelkuhn committed:
stage: fix resources being deleted from local cache (#10152)

Non-expanded resources were being deleted from the cache on account
of two behaviors:

* ResourceStage was moving files rather than copying them, and uses
  "os.path.realpath" to resolve symlinks
* CacheFetchStrategy creates a symlink to a cached resource rather
  than copying it

This alters the first behavior: ResourceStage now copies the file
rather than moving it.

balay committed:
PETSc: add version 3.10.3 (#10154)

adamjstewart committed:
Restrict patch to oldest version of libbsd (#10145)

adamjstewart committed:
Add latest version of docbook-xsl (#10147)

adamjstewart committed:
Add list_url to docbook-xml (#10146)

adamjstewart committed:
Add missing libiconv dependency to libxml2 (#10148)

adamjstewart committed:
Add python bindings to libxslt (#10149)

Commit:
Updated Dakota to 6.9 (#10157)

* Updated Dakota to 6.9

* Change comment wording

Commit:
phist tag 1.7.5 was moved (#10155)

michaelkuhn committed:
fetch_strategy: improve GitFetchStrategy (#10153)

Currently, only the Git repsository's URL shows up in the `spack info`
output, which makes it hard to distinguish different versions:
```
Safe versions:
    develop    [git] https://github.com/LLNL/H5Z-ZFP.git
    0.8.0      [git] https://github.com/LLNL/H5Z-ZFP.git
    0.7.0      [git] https://github.com/LLNL/H5Z-ZFP.git
```

This change adds additional information as shown when cloning a Git
repository:
```
Safe versions:
    develop    [git] https://github.com/LLNL/H5Z-ZFP.git on branch master
    0.8.0      [git] https://github.com/LLNL/H5Z-ZFP.git at commit af165c4
    0.7.0      [git] https://github.com/LLNL/H5Z-ZFP.git at commit 58ac811
```

odoublewen committed:
updated checksum for kallisto (#10158)

michaelkuhn committed:
sqlite: add 3.26.0 (#10138)

mwkrentel committed:
dyninst: patch to build dyninst with older gcc (#10160)

* dyninst: patch to build dyninst with older gcc

Add 'v9.3.2-auto.patch'.  This patch changes some 'auto t: type' usage
to the older but equivalent 'type.begin()'.  This allows building
dyninst 9.3.2 with gcc 4.4 which doesn't support the newer syntax.
This patch is harmless with newer gcc.

* Fix typo in patch.  (My bad for sloppy cut-and-paste.)

* Restrict the patch to gcc 4.7 and earlier.  gcc 4.8 supports the newer
usage and thus doesn't need the patch.

gsjaardema committed:
CGNS: Allow use of git branch versions (#10156)

* CGNS: Allow use of git branch versions

Add ability to use git `master` and `develop` branches of the CGNS library.  The `seacas` package sometimes needs to use the `master` or `develop` branch due to some new changes to the parallel I/O implementation in CGNS which has not yet been released.

* CGNS: Reorder to put newest version on top

Based on review comments, place newest version (develop) first in list.

Commit:
packages/phist: add maintainer (#10162)

Commit:
fix multiple issues with the docker images (#9718)

 - fixed an issue where some undesirable parts of
   the spack source tree were being copied into
   the image context.

 - added a workaround for a tty ioctl warning on
   ubuntu

 - adjusted how the main images are built so that
   `RUN spack ...` works automatically for child
   images that base themselves on them.

Commit:
Fix docker builds (#9805)

* move docker test logic to share/spack/qa
* update Dockerfile for archlinux

olesenm committed:
OpenFOAM-v1812 - new version (#10167)

thelfer committed:
Update TFEL package (#9995)

* add the tfel package

* fix the tfel package

* fix the tfel package

* fix the tfel package

* Taking Adam J. Steward' remarks into account

* fixes trailing white spaces

* Update description

* Update dependencies following @adamjstewart adices

* Style fixes

* Style fixes

* Add java optional support

* add the maintainers attribute (following @alalazo advice), disable interface not selected (following @adamjstewart advice)

* flake8 fixes

* Fix Cast3M and python-bindings support. Python detection is made compatible with cmake'FindPythonLibs module (at least how it is used in TFEL)

* Style fixes

* Style fixes

* Fix test on python version

* Follow @adamjstewart advices: code is much cleaner and readable

* Small fix

* Small fix

* Add comment

* Small fix in cmake option

* try again (trying to overcome Travis CI unstable build process)

s-sajid-ali committed:
update h5py to 2.9.0 (#10169)

* 	modified:   var/spack/repos/builtin/packages/py-h5py/package.py

* Update package.py

* Update package.py

DarylGrunau committed:
R3D spackage (#10164)

* r3d spackage

* address Adam's review comments

* fix flake8: too many blank lines

gonsie committed:
Remove /nfs/tmp2 from default build_stage locations (#10170)

* Remove /nfs/tmp2 from default configuration
* /nfs/tmp2 is going away from LC... and doesn’t exist for the rest of the world.
* update documentation to remove /nfs/tmp2 as well

tgamblin committed:
repo: refactor indexes to be more extensible

- virtual provider cache and tags were previously generated by nearly
  identical but separate methods.

- factor out an Indexer interface for updating repository caches, and
  provide implementations for each type of index (TagIndex,
  ProviderIndex) so that more can be added if needed.

- Among other things, this allows all indexes to be updated at once.
  This is an advantage because loading package files is the real
  overhead, and building the indexes once the packages are loaded is
  trivial. We avoid extra bulk read-ins by generating all package indexes
  at once.

- This can be extended for dependents (reverse dependencies) and patches
  later.

hartzell committed:
Add libiconv variant to glib (#10165)

Enable explicitly building glib with or without libiconv.

Heads up: if you build with ~iconv but there is something else in the
spec that includes libiconv then glib will *still* automagically
discover it and the build will fail.

matz-e committed:
py-numba: fix py-llvmlite version (#10175)

j-ogas committed:
add charliecloud 0.9.6 (#10176)

baberlevi committed:
gpgme: new package (#10174)

gartung committed:
Merge remote-tracking branch 'upstream/develop' into fnal-develop-merge

balay committed:
petsc-3.10.3: syncs with superlu-dist-6.1. (#10166)

Commit:
r-corhmm: new package at 1.22 (#9859)

Commit:
Add missing module env variable setting causing OSS to fail. (#10182)

tgamblin committed:
patches: add a per-repository patch index

- this fixes a bug where if we save a concretized sug-DAG where a package
  had been patched by a dependent, and the dependent was not in the DAG,
  we would not read in all patches correctly.

- Rather than looking up patches in the DAG, we look them up globally
  from an index created from the entire repository.

- The patch cache is a bit tricky for several reasons:

  - we have to cache information from packages, specifically, the patch
    level and working directory.

  - FilePatches need to know which package owns them, so that they can
    figure out where the patch lives.  The repo can change locations from
    run to run, so we have to store relative paths and restore them when
    the cache is reloaded.

  - Patch files can change underneath the cache, because repo indexes
    only update on package changes.  We currently punt on this -- there
    are stub methods for needs_update() that will need to check patch
    files when packages are loaded.  There isn't an easy way to do this
    at global indexing time without making the FastPackageChecker a lot
    slower.  This is TBD for a future commit.

  - Currently, the same patch can only be used one way in a package. That
    is, if it appears twice with different level/working_dir settings,
    bad things will happen.  There's no package that current uses the
    same patch two different ways, so we've punted on this as well, but
    we may need to fix this in the future by moving a lot of the metdata
    (level, working dir) to the spec, and *only* caching sha256sums in
    the PatchCache.  That would require some much more complicated tweaks
    to the Spec, so we're holding off on that til later.

- This required patches to be refactored somewhat -- the difference
  between a UrlPatch and a FilePatch is still not particularly clean.

tgamblin committed:
tests: add a test to make sure that patched specs can be round-tripped

- previously, if a concrete sub-DAG with patched specs was written out
  and read back in, its patches would not be found because the dependent
  that patched it was no longer in the DAG.

- Add a test to ensure that the PatchCache handles this case.

- Also add tests to ensure that patch objects are properly created from
  Specs -- previously we only checked that the patches were on the Spec.

neilflood committed:
py-pynio: new package (#10074)

* py-pynio: new package

* Major re-working to add some of the optional dependencies, and make it conform to the approach described in the pynio setup.py comments. Much of it turned out to be too complicated, and I have added some notes in comments to explain this to future adventurers.

* Added variants for the hdf5 and gdal dependencies, which do seem to be genuinely optional. Also added absolute dependencies for jpeg and zlib, which setup.py says it wants, but were being found as the os-installed versions.

* Added descriptions for the new variants, and moved them to above the dependencies

bvanessen committed:
Feature add breathe and m2r (#10180)

* Added a package for the MDAnalysis toolkit.

* Added python packages for breathe and m2r.  Updated versions and URL
for the sphinx package.

* Fixed flake8 error

* Updated package requirements based on feedback.

* Updated with requirements from the setup.py

* Switched it back to pypi.

sbna committed:
openFOAM-paraview-catalyst (#10129)

* paraview: adding variants to use external packages as internal do not compile

* paraview: add latest paraview version

* catalyst: fixed libvtkexpat undefined reference linking error in Catalyst 5.5

* catalyst: add latest catalyst version

* catalyst: added ParaView_DIR env variable to catalyst module

* add paraview, catalyst patches

- https://gitlab.kitware.com/vtk/vtk-m/merge_requests/1166
- https://gitlab.kitware.com/paraview/paraview/merge_requests/2433
- https://gitlab.kitware.com/paraview/paraview/merge_requests/2436

* - Handle updated library paths for catalyst.

  Versions 5.4 and old places libraries under a paraview subdirectory.
  Eg, "lib/paraview-5.4", they are now placed directly under "lib"

- Minor code style cleanup

* Handle update library and python paths for ParaView-5.5

* catalyst: added ParaView_DIR path to spack_env

* BUG: applied the patch to the extracted catalyst source files

* paraview: added missing self to a member variable

* paraview: added Paraview_DIR to env

* catalyst: added osmesa variant

* of-catalyst: added new package

* add (FOAM,WM)_PROJECT_DIR  also to spack_env environment

* depends on first openfoam release supporting catalyst

* openfoam-com: added missing env variables to module generation

* openfoam: fixed flake8 errors

* of-catalyst: added full variant and openfoam version dependency

* paraview: adding variants to use external packages as internal do not compile

* catalyst: fixed libvtkexpat undefined reference linking error in Catalyst 5.5

* catalyst: added ParaView_DIR env variable to catalyst module

* add paraview, catalyst patches

- https://gitlab.kitware.com/vtk/vtk-m/merge_requests/1166
- https://gitlab.kitware.com/paraview/paraview/merge_requests/2433
- https://gitlab.kitware.com/paraview/paraview/merge_requests/2436

* - Handle updated library paths for catalyst.

  Versions 5.4 and old places libraries under a paraview subdirectory.
  Eg, "lib/paraview-5.4", they are now placed directly under "lib"

- Minor code style cleanup

* Handle update library and python paths for ParaView-5.5

* catalyst: added ParaView_DIR path to spack_env

* BUG: applied the patch to the extracted catalyst source files

* of-catalyst: added new package

* add (FOAM,WM)_PROJECT_DIR  also to spack_env environment

* depends on first openfoam release supporting catalyst

* paraview: added missing self to a member variable

* openfoam-com: added missing env variables to module generation

* openfoam: fixed flake8 errors

* paraview: added Paraview_DIR to env

* catalyst: added osmesa variant

* of-catalyst: added full variant and openfoam version dependency

* paraview-catalyst: use always external expat and netcdf

* of-catalyst: reformatted package description

* paraview-catalyst: removed duplicated function

* catalyst: fixed flake8 error

* of-catalyst: fixed license header

* of-catalyst: minor changes

* of-catalyst: renamed gitrepo with git

* of-catalyst: removed useless gitrepo parameter

neilflood committed:
py-cftime: new package (#10066)

* py-cftime: new package

* Added dependency on py-setuptools, and version range for py-cython, thanks to the ever-vigilant @adamjstewart

balay committed:
xsdk-0.4.0: add/update package versions: (#10038)

* xsdk-0.4.0: add/update package versions:
  hypre@2.15.1
  mfem@3.4.0
  superlu-dist@6.1.0
  trilinos@12.14.0-rc1
  petsc@3.10.3
  dealii@9.0.1
  pflotran@xsdk-0.4.0
  alquimia@xsdk-0.4.0
  sundials@3.2.1
  plasma@18.11.1
  magma@2.4.0
  amrex@18.10.1
  slepc@3.10.1
  omega-h@9.19.1
  strumpack@3.1.1
  pumi@2.2.0
  tasmanian@6.0
  phist@1.7.5
  dtk-3.0 [via trilinos features '+dtk+intrepid2+shards']

add option to disable omega-h build [wrt gcc-8 conflict]
  spack install xsdk~omega-h
add option to disable dealii build [wrt cori build error]
  spack install xsdk~dealii

* trilinos: default version should be the prior release

Commit:
travis: fix pip install for docker stage (#10177)

tgamblin committed:
bugfix: handle unicode properly in spack.util.executable (#10186)

- When returning string output, use text_type and decode utf-8 in Python
  2 instead of using `str`

- This properly handles unicode, whereas before we would pass bad strings
  to colify in `spack blame` when reading git output

- add a test that round-trips some unicode through an Executable object

erimar77 committed:
manta: Add version 1.5.0 and add missing zlib dependency (#10178)

tgamblin committed:
ProviderIndex uses json instead of YAML

- indexes should use json, not YAML, to optimize for speed
- only use YAML in human-editable files
- this makes ProviderIndex consistent with other indexes

tgamblin committed:
commands: add `spack resource` command to inspect downloadable files

- currently just looks at patches
  - allows you to find out which package applied a patch to a spec

- intended to work with tarballs and resources in the future.

- add tab completion for `spack resource` and subcommands

adamjstewart committed:
Add PyTorch 1.0 (#10193)

adamjstewart committed:
Add latest version of NCCL (#10194)

davydden committed:
dealii: fix build with some Intel compilers (#10188)

davydden committed:
environment: ignore invalid files names in var/spack/environments (#10198)

alalazo committed:
tests: run 'test_compiler_add' serially in Python 2 (#10127)

* Don't spawn processes during python 2.6/2.7 'test_compiler_add' unit tests
* Travis seems to choke on the excessive parallelism in `compiler find`

balay committed:
slepc: fix homepage URL (#10199)

junghans committed:
mawk: added 1.3.4_p20171017 (#10197)

adamjstewart committed:
Fix libxml2+python and libxslt+python import tests (#10196)

* Fix libxml2+python import tests

* Same for libxslt package

adamjstewart committed:
Fix pkgconf downloads (#10202)

junghans committed:
pgmath: depend on awk (#10200)

StackKorora committed:
Add Stata package (#10189)

* First draft for installing Stata 15

I've tested this works on my systems and gives a successful install of STATA 15.

* Flake8 cleanup

* Another Flake8

* Update license header

* Various improvements

A lot of improvements as suggested by @adamjstewart

* flake8 check

likask committed:
fix moab installation for version 5.0.2 removed from ftp (#10201)

* moab install from git repository, fix version 5.0.2 removed from ftp server

* add moab dependencies above verson 5.0.1

Commit:
xgboost: new package at 0.81 (#9854)

AndrewGaspar committed:
Add new versions of optional-lite and string-view-lite (#10179)

tgamblin committed:
bugfix: nested directives were broken in python 3

- The nested directive implementation was broken for python 3

- directive results were not properly removed from the directive list
  when it was processed in the DirectiveMeta metaclass.

- the issue was that remove_directives only descended into a list or
  tuple, but in Python3, the initial value passed to the function is a
  view of dictionary values.

- make it a list to fix things, and add a regression test.

erimar77 committed:
py-testinfra: Add version 1.18.0 (#10210)

erimar77 committed:
velvet: Add missing zlib dependency to fix installation errors (#10212)

erimar77 committed:
salmon: Add version 0.12.0 (#10211)

adamjstewart committed:
Add GDAL 2.4.0 (#10215)

tgamblin committed:
Use homebrew addon instead of manually invoking brew (#10213)

tgamblin committed:
bugfix: add codecov upload back to travis.yml (#10220)

Codecov upload was accidentally removed in #9805

tgamblin committed:
coverage: use kcov to get coverage for our cc script

hartzell committed:
Always build glib with iconv (#10219)

* Always build glib with iconv

My early PR, #10165, which added a variant to configure glib to use
libiconv and defaulted to false, seems to be causing more trouble than
the knob is worth.

This changes the glib package to always depend on and use libiconv.

* libiconv depends_on is no longer conditional

tgamblin committed:
stopgap: allow travis to fail for Python 2.6 again

- Travis still fails on `spack spec` in Python 2.6.
- Allow the failure until we get to the bottom of it.

davydden committed:
dealii: workaround concretization issue with PETSc (#10217)

Sinan81 committed:
xgboost: make sure "libxgboost.so" is provided under prefix/lib (#10208)

naromero77 committed:
Over specifying Quantum Espresso compiler variables causes problems in some corner cases. (#10216)

goxberry committed:
py-sphinxautomodapi: new python package (#10226)

gklimowicz committed:
flang, llvm, pgmath: Add release for 2018-09-21 (#10224)

Commit:
ds: new package at 9.8.0 (#9852)

* ds: new package at 9.8.0rc6

* ds: move patches to patch func

* ds: use built-in configure phase, remove autoreconf skip

* ds: revert to stable 9.8.0

* ds: remove unnecessary dep types

* ds: make url consistent with versions

hartzell committed:
bamutil builds with newer gcc's (#10229)

The most recent release of bamutil that we support uses an embedded
copy of libStatGen that has several issues that keep it from building
with newer releases of gcc.

They've all been fixed upstream and the latest release of bamutil
would pick them up if/when we support it.  The build process has
changed though, plus my team needs *this* version.

This commit backports those fixes.

hartzell committed:
fastx-toolkit builds with newer gcc's (#10228)

There's a missing break in a switch statement that newer gcc's
dislike.

Our #4696 simply disallowed newer gcc's.

This fixes the problem instead.

It's been [PR'ed upstream](https://github.com/agordon/fastx_toolkit/pull/22).

Tested with gcc@5.5.0 and gcc@8.2.0 on CentOS.

hartzell committed:
aspell builds with newer gcc's (#10227)

Apply fix from aspell issue (519) for a pointer dereference bug that
newer versions of gcc won't let slip past.

There hasn't been a release that includes the fix, this applies the
change to the latest release.

hartzell committed:
express: new version, use tags and fix gcc@6.0.0: (#10230)

* express: new version, use tags and fix gcc@6.0.0:

Express fails to build with gcc@6.0.0:.

The debian project [has a
fix](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=811859) but
they don't seem to have pushed it upstream.

I've opened an issue and a PR in eXpress repo, but eXpress isn't
actively developed, so I'm fixing it here too.

Since the Spack package was created, the eXpress team tagged their
releases.  I've updated the package to use the tags.

Version 1.5.1 used to be known as 2015-11-29 (same commit).  1.5.2 is
new(er).

* Make flake8 happ{y,ier}

tgamblin committed:
license: license command prints sorted, non-redundant results

- spack license command now ignores symlinks

- spack license list-files now prints sorted output, and checks for files
  we've already seen.

tgamblin committed:
copyright: update license headers for 2013-2019 copyright.

tgamblin committed:
flake8: fix flake8 issues with invalid escapes

- in many files, regular strings were used in places where raw strings
  should've been used.

- convert these to raw strings and get rid of new flake8 errors

tgamblin committed:
README: LLNL release number and SPDX id don't need to be monospaced.

alalazo committed:
Improve validation of modules.yaml (#9878)

This PR improves the validation of `modules.yaml` by introducing a custom validator that checks if an attribute listed in `properties` or `patternProperties` is a valid spec. This new check applied to the test case in #9857 gives:

```console
$ spack install szip
==> Error: /home/mculpo/.spack/linux/modules.yaml:5: "^python@2.7@" is an invalid spec [Invalid version specifier]
```

Details: 
* Moved the set-up of a custom validator class to spack.schema
  * In Spack we use `jsonschema` to validate configuration files 
    against a schema. We also need custom validators to enforce
    writing default values within "properties" or "patternProperties"
    attributes.

  * Currently, validators were customized at the place of use and with the
    recent introduction of environments that meant we were setting-up and
    using 2 different validator classes in two different modules.

  * This commit moves the set-up of a custom validator class in the
    `spack.schema` module and refactors the code in `spack.config` and
    `spack.environments` to use it.

* Added a custom validator to check if an attribute is a valid spec
  * Added a custom validator that can be used on objects, which yields an
    error if the attribute is not a valid spec.

* Updated the schema for modules.yaml

* Updated modules.yaml to fix a few inconsistencies:
  - a few attributes were not tested properly using 'anyOf'
  - suffixes has been updated to also check that the attribute is a spec
  - hierarchical_scheme has been updated to hierarchy

* Removed $ref from every schema
  * $ref is not composable or particularly legible
  * Use python dicts and regular old variables instead.

hartzell committed:
Add list_url for r-readr, finding old versions (#10231)

r-tidyverse needs r-readr@1.1.1, r-reardr needs a list_url so that it
can find it.

benzwick committed:
FEniCS: workaround concretization issue with PETSc (#10235)

alalazo committed:
Removed an extra comma after a dict literal (#10236)

The extra comma was turning a dict into a tuple, and caused errors
during validation of the schema.

gsjaardema committed:
PARALLEL-NETCDF: Update new version and location (#10172)

* PARALLEL-NETCDF: Update new version and location

PnetCDF-1.11.0 is released.

Also, the canonical download area has been changed and they are now using git, so can also provide a develop and master checkout.
One issue is that they changed the name of the tar files, so 1.11.0 needs special handling (and future versions will also).

All checksums at new location match the checksums from the old location.

* Address concerns in review

Added a `url_for_version` function to handle the name change in the tar files at version 1.11.0 and later.  Updated description to match current shown on website.  Updated `homepage` setting since it has recently moved.

* Address issues in second review

benzwick committed:
slepc4py: new versions (up to 3.10.1) and update website (#10239)

benzwick committed:
petsc4py: new versions (up to 3.10.0) and update website (#10238)

ax3l committed:
Catch2: v2.5.0 (#10237)

Add the 2.5.0 release of catch2.

michaelkuhn committed:
patch: split up fetch and clean into separate methods (#10150)

"mirror create" was invoking a package's do_patch method in order to
retrieve and archive URL patches. If a package implements a "patch"
method, this is also called as part of do_patch; this failed when the
package-specific implementation referred to environment variables
that are only available at the time the package is built
(e.g. "spack_cc").

This change introduces fetch and clean methods for patches. They are
no-ops for FilePatch but perform the appropriate actions for
UrlPatch. This allows "mirror create" to invoke do_fetch, which does
not call the package's patch method.

benzwick committed:
Add missing code block (#10243)

Otherwise the `--` in `--dot` is rendered as a single en-dash.

neilflood committed:
openssl: add location of system certificates for SUSE Linux (#10242)

scheibelp committed:
Add tests for expanding/non-expanding resources

jcftang committed:
Fix download path, the existing one has been removed from nvidia's site (#10253)

Commit:
Added package osmctools (#10249)

* Added package osmctools

* Simplified

Removed functions that did the same as existing defaults

* flake8

michaelkuhn committed:
mvapich2: add missing zlib dependency (#10254)

alalazo committed:
Multi-valued variants: better support for combinations (#9481)

This enforces conventions that allow for correct handling of
multi-valued variants where specifying no value is an option,
and adds convenience functionality for specifying multi-valued
variants with conflicting sets of values. This also adds a notion
of "feature values" for variants, which are those that are understood
by the build system (e.g. those that would appear as configure
options). In more detail:

* Add documentation on variants to the packaging guide
* Forbid usage of '' or None as a possible variant value, in
  particular as a default. To indicate choosing no value, the user
  must explicitly define an option like 'none'. Without this,
  multi-valued variants with default set to None were not parsable
  from the command line (Fixes #6314)
* Add "disjoint_sets" function to support the declaration of
  multi-valued variants with conflicting sets of options. For example
  a variant "foo" with possible values "a", "b", and "c" where "c"
  is exclusive of the other values ("foo=a,b" and "foo=c" are
  valid but "foo=a,c" is not).
* Add "any_combination_of" function to support the declaration of
  multi-valued variants where it is valid to choose none of the
  values. This automatically defines "none" as an option (exclusive
  with all other choices); this value does not appear when iterating
  over the variant's values, for example in "with_or_without" (which
  constructs autotools option strings from variant values).
* The "disjoint_sets" and "any_combination_of" methods return an
  object which tracks the possible values. It is also possible to
  indicate that some of these values do not correspond to options
  understood by the package's build system, such that methods like
  "with_or_without" will not define options for those values (this
  occurs automatically for "none")
* Add documentation for usage of new functions for specifying
  multi-valued variants

cdwdirect committed:
doxygen: URL and version hashes updated to project's GitHub home (#10250)

* Update package.py

Doxygen has migrated from a private SVN to GitHub. This PR updates the URLs and adds versioned commit hashes from GitHub. It also includes version 1.8.15 as the latest option.

* Update package.py

gartung committed:
Merge remote-tracking branch 'upstream/develop' into fnal-develop-merge

hartzell committed:
bowtie: new versions and %gcc@8.0.0: fix (#10232)

* bowtie: new versions and %gcc@8.0.0: fix

Bowtie development shifted from Sourceforge to GitHub.

This commit adds several versions from GitHub, using the archive
tarballs.  Note that the URL for 1.2.2 tarball is actually an '_p1'
tag....

It leaves the old 1.2 release download info in place.

Versions 1.2.0 and newer come from GitHub (I'm unsure if the 1.2 and
1.2.0 are equivalent).

Include a fix that enables %gcc@8.0.0: from:

  https://github.com/BenLangmead/bowtie/issues/87

But, v1.2.2 has trouble with "newer" gcc's, so it only adds v1.2.2 for
%gcc@6.0.0:.

Feel free to tighten this.  I know that:

- 1.2 -> 1.2.2 work with %gcc@5.5.0;
- 1.2 -> 1.2.1.1 work with %gcc@8.2.0; and
- 1.2.2 fails with %gcc@8.2.0

* Tighten to `conflicts('%gcc@8:', when='@1.2.2')`

* Point 1.2.2 and 1.2.2_p1 at the 1.2.2_p1 tarball

scheibelp committed:
For spec-file-based installs, store the initial spec read from the file as the abstract spec

healther committed:
new package: py-webencodings (#10277)

Change-Id: Icf20035367a34fa88acfef2d3fa3742c3f3901ac

michaelkuhn committed:
go: add 1.11.4 (#10272)

erimar77 committed:
salmon: Add version 0.12.0 (#10271)

rgmiller committed:
Check for ARM compiler in addition to clang (#10280)

Add "-rtlib=compiler-rt" to CFLAGS when building with either Clang or
the ARM compiler.

Refs #10279

healther committed:
new version 1.0.1 for py-html5lib (#10278)

Change-Id: I2837dc098fce142b152d2969687d1873e182bbc8

erimar77 committed:
libzip: Add zlib dependency (#10270)

* libzip: Add zlib dependency

* libzip: Add version limit

dorier committed:
added openpa package (#10265)

odoublewen committed:
updated versions and depends_on() (#10264)

odoublewen committed:
updated versions (#10263)

odoublewen committed:
r-dt: new versions (#10261)

* updated release and checksums, also tightened depends_on()

* remove blank line at end of file

* removing new version 0.5

Commit:
Added bash 5.0 (#10282)

erimar77 committed:
usearch: Make installed binary executable (#10269)

* usearch: Make binary executable

* usearch: remove join_path

thelfer committed:
New package: MGIS (#10273)

* add the tfel package

* fix the tfel package

* fix the tfel package

* fix the tfel package

* Taking Adam J. Steward' remarks into account

* fixes trailing white spaces

* Update description

* Update dependencies following @adamjstewart adices

* Style fixes

* Style fixes

* Add java optional support

* add the maintainers attribute (following @alalazo advice), disable interface not selected (following @adamjstewart advice)

* flake8 fixes

* Fix Cast3M and python-bindings support. Python detection is made compatible with cmake'FindPythonLibs module (at least how it is used in TFEL)

* Style fixes

* Style fixes

* Fix test on python version

* Follow @adamjstewart advices: code is much cleaner and readable

* Small fix

* Small fix

* Add comment

* Small fix in cmake option

* try again (trying to overcome Travis CI unstable build process)

* Add support for the MFrontGenericInterfaceSupport project (MGIS)

* Style fixes

* Package documentation update

* Package documentation update

* Fix a typo thanks to Andreas Baumbach review

* Follow Adam J. Stewart advices

* Fix type

odoublewen committed:
r-dbplyr: new versions (#10262)

* added new versions, and tightened up depends_on() specs

* added type=('build', 'run') to r dependency

odoublewen committed:
r-data-table: new versions (#10260)

* updated versions for r-data-table

* added type=('build', 'run') to r dependency

odoublewen committed:
r-cowplot: new versions, more specific dependencies (#10259)

* updated versions for r-cowplot

* depends_on: added type=('build', 'run') to r, and added r-scales

odoublewen committed:
r-ape: new versions (#10258)

* updated versions for r-ape

* added type=('build', 'run') to r dependency

Commit:
Armcompiler (#9840)

* Initial compiler support

* added arm.py

* Changed licence to Arm suggested header

* Changed licence to the same as clang.py
Main author of file is Nick Forrington <Nick.Forrington@arm.com>
Minor changes by Srinath Vadlamani <srinath.vadlamani@arm.com>

* compilers: add arm compiler detection to Spack

- added arm.py with support for detecting `armclang` and `armflang`

Co-authored-by: Srinath Vadlamani <srinath.vadlamani@arm.com>

* Changed to using get get_compiler_version

* linking to general cc for arm compiler

* For arm compiler add CFLAGS to use compiler-rt rtlib.

* Escape for special characters in rexep

* Cleaned up for Flake8 to pass.

* libcompiler-rt should be part of the LDFLAGS not CFLAGS

* fixed m4 when using clang to used LDFLAGS.  Fixed comments for arm.py to display compiler --version output with # NOAQ for flakes pass.

* added arm compilers

* proper linked names

odoublewen committed:
bug fix: copy permissions when staging (#10285)

Fixes #10284

#10152 replaced shutil.move with llnl's copy and copy_tree for
resources. This did not copy permissions so led to later failures
if an executable was copied (e.g. a configure script). This uses
install/install_tree instead, which preserve permissions.

gartung committed:
replace placeholder and origin install path

Commit:
Updated fish to 3.0.0 (#10286)

ax3l committed:
OpenMPI: Add Development Version (#10257)

Add the development version of OpenMPI from the git repo
in `master`.

gartung committed:
check for prefix in placeholder substituted binaries

gartung committed:
autopep8

gartung committed:
Merge remote-tracking branch 'origin/buildcache-placeholder-fix' into fnal-develop-merge

gartung committed:
Merge branch 'buildcache-placeholder-fix' into fnal-develop-merge

greenc-FNAL commented:
@gartung Let me know when you think this is ready for merge.

gartung commented:
Its is ready
